src/ripper/ripcddialog.cpp

```diff
@@ -293,20 +291,12 @@ void RipCDDialog::DeviceSelected(int device_index) {
   }
 
   connect(cdda_device_.get(), SIGNAL(DiscChanged()), SLOT(DiscChanged()));
-
-  // get SongLoader from device and connect signals
-  loader_ = cdda_device_->loader();
-  Q_ASSERT(loader_);
-
-  connect(loader_, SIGNAL(SongsDurationLoaded(SongList)),
+  connect(cdda_device_.get(), SIGNAL(SongsDiscovered(SongList)),
           SLOT(BuildTrackListTable(SongList)));
-  connect(loader_, SIGNAL(SongsMetadataLoaded(SongList)),
-          SLOT(UpdateTrackListTable(SongList)));
-  connect(loader_, SIGNAL(SongsMetadataLoaded(SongList)),
-          SLOT(AddAlbumMetadataFromMusicBrainz(SongList)));
+  connect(cdda_device_.get(), SIGNAL(SongsDiscovered(SongList)),
+          SLOT(SetAlbumMetadata(SongList)));
```

Comment:
Would it make sense to connect SongsDiscovered to a single slot?

Reply:
We could do that, but I think the logical separation of updating the track list and updating the metadata fields makes some sense as well and helps readability of the code. We could call `BuildTrackListTable` and `SetAlbumMetadata` from inside a new handler for SongsDiscovered, to avoid the multiple connections, though.